torch/_dynamo/skipfiles.py

```diff
@@ -145,78 +157,109 @@ def _module_dir(m: types.ModuleType):
+    "torch.distributed._tensor.api",
+    "torch.distributed._tensor.device_mesh",
+    "torch.distributed.tensor.parallel._data_parallel_utils",
+    "torch.distributed.tensor.parallel._utils",
+    "torch.distributed.tensor.parallel.style",
+    "torch.nn.modules.container",
+    "torch.optim._functional",
+    "torch.random",
+    "torch.utils._content_store",
+    "torch.utils._foreach_utils",
 }
 
+
 if torch.distributed.is_available():
     # Inline the checkpoint code from distributed
-    import torch.distributed.algorithms._checkpoint.checkpoint_wrapper
-
-    FILENAME_INLINELIST |= {
-        torch.distributed.algorithms._checkpoint.checkpoint_wrapper.__file__
-    }
+    FILE_INLINELIST |= {"torch.distributed.algorithms._checkpoint.checkpoint_wrapper"}
 
 # Include optimizer code for tracing
-FILENAME_INLINELIST |= {
-    inspect.getfile(obj)
-    for obj in torch.optim.__dict__.values()
-    if inspect.isclass(obj)
-}
-
-# TODO (zhxchen17) Make exportdb importable here.
-FILENAME_INLINELIST |= set(
-    glob.glob(_module_dir(torch) + "_export/db/examples/*.py"),
-) | {
-    _module_dir(torch) + "_export/wrappers.py",
+FILE_INLINELIST |= {
+    str(obj.__module__) for obj in torch.optim.__dict__.values() if inspect.isclass(obj)
 }
 
 
+# TODO: consolidate SUBMODULE_INLINELIST and FILE_INLINELIST into one list
 # Force inline functions under these modules, even the modules is in *_SKIPLIST.
 SUBMODULE_INLINELIST = {
-    torch.nn,
-    torch.distributions,
-    torch.testing,
-    torch.ao.nn,
-    torch._refs,
-    torch._prims,
-    torch._decomp,
-    torch.utils._contextlib,
-    torch.utils._pytree,
-    torch.fx._pytree,
-    torch.sparse,
+    "torch._refs",
+    "torch._prims",
+    "torch._decomp",
+    "torch.ao.nn",
+    "torch.distributions",
+    "torch.fx._pytree",
+    "torch.nn",
+    "torch.sparse",
+    "torch.testing",
+    "torch.utils._contextlib",
+    "torch.utils._pytree",
 }
 
 
 if torch.distributed.is_available():
-    from torch.distributed import _functional_collectives
+    SUBMODULE_INLINELIST.add("torch.distributed._functional_collectives")
+
+
+@functools.lru_cache(None)
+def get_func_inlinelist():
+    inlinelist = set()
+    for f in FUNC_INLINELIST:
+        module_name, fn_name = f.rsplit(".", 1)
+        m = importlib.import_module(module_name)
+        fn = getattr(m, fn_name)
+        inlinelist.add(fn.__code__)
+    return inlinelist
+
+
+@functools.lru_cache(None)
+def get_file_inlinelist():
+    inlinelist = set()
+    for f in FILE_INLINELIST:
+        inlinelist.add(_module_dir(torch) + f[len("torch.") :].replace(".", "/"))
+    return inlinelist
+
+
+@functools.lru_cache(None)
+def get_submodule_inlinelist():
+    inlinelist = set()
+    for m in SUBMODULE_INLINELIST:
+        inlinelist.add(_module_dir(torch) + m[len("torch.") :].replace(".", "/"))
+    return inlinelist
 
-    SUBMODULE_INLINELIST.add(_functional_collectives)
+
+if TYPE_CHECKING:
```

Comment:
I don't really understand what your intention was here. TYPE_CHECKING is only true when you are doing something like mypy type checking, but you won't run any of the code here in that case...

Reply:
If you want just a plain test case, probably easiest to just define a test case that runs this code.

Reply:
I want to use this to check the functions/modules in the list can be imported or loaded correctly in case there is typo. I agree unit test is a better place and moved it there.